dhis-2/dhis-api/src/main/java/org/hisp/dhis/webapi/controller/event/webrequest/PagingAndSortingCriteriaAdapter.java

```diff
@@ -76,10 +77,15 @@
 
     public boolean isPagingRequest()
     {
-        return paging != null && paging ||
-            pageSize != null ||
-            page != null ||
-            totalPages;
+        return !isSkipPaging() && (
+                paging != null && paging ||
+                        pageSize != null ||
+                        page != null ||
+                        totalPages);
     }
 
+    private boolean isSkipPaging() {
```

Comment:
It wouldn't be the same as set skipPaging default value to false?

I am asking just because you went this way for pageSize and if it is the same case it would be good to maintain consistency

Reply:
I was tempted to do so, but this code is shared between all other endpoints, and I don't really understand why we need a flag to skipPaging explicitrly, while we have "paging=true|failse" which might have the same meaning. I really think we should review the meaning of these parameters and introduce fair defaults in a centralized way. I was thinking to create a small ticket to refactor these parameters for new tracker, and updating documentation, that's why I didn't put false in this ticket, I'd rather spend more time thinking about it and how to avoid endpoint specific conditions on the paging parameters, so for now this method is safer than using false as default.

Reply:
👍